src/engine/renderer/gl_shader.cpp

```diff
@@ -770,9 +770,29 @@ std::string     GLShaderManager::BuildGPUShaderText( Str::StringRef mainShaderNa
 	// so we have to reset the line counting.
 	env += "#line 0\n";
 
-	std::string shaderText = env + libs + GetShaderText(filename);
+	std::string shaderText = env + libs + GetShaderText( filename );
 
-	return shaderText;
+	std::istringstream shaderTextStream( shaderText );
+	std::string shaderMain;
+
+	std::string line;
+
+	while ( std::getline( shaderTextStream, line, '\n' ) ) {
+		std::string::size_type position = line.find( "#insert" );
```

Comment:
This shouldn't match anywhere in the line, only at the beginning (maybe preceded by whitespace). It would be annoying if commenting it out like `//#insert blah` didn't work.

Reply:
I agree on both. The extensions thing is something that's used throught gl_shaders.cpp IIRC, so changing all of those should probably be its own pr.

Reply:
Now it will only match if `#insert` is directly following whitespace characters or at the start of the line, no more fixed suffix as well. I've left the extension there because that's what we use everywhere, so having it suddenly different in one place would be confusing, plus I don't really see the benefit of dropping the extension if we're gonna put all shader code into `.glsl` files anyway.